angr/knowledge_plugins/variables/variable_manager.py

```diff
@@ -1213,6 +1214,7 @@ def _is_variable_only_used_by_phi_stmt(
             # not used at all?
             return False
         for acc in accesses:
+            assert acc.location.block_addr is not None
```

Comment:
It is more idiomatic to use typing.cast I think

Reply:
I feel their semantics are different? Assertions imply that the asserted conditions must be true, otherwise something is horribly wrong and should abort execution, while cast() does not provide any execution-related benefits.

Reply:
Speaking of which, we should differentiate between CodeLocation and ExternalCodeLocation and be done with this assert block_addr is not None nonsense for good… will refactor it at a later time.

Reply:
In python, casting is not reinterpretation of data but rather just an escape hatch in the static type system. If you want runtime checking I think a custom exception is preferable to an assert.

Reply:
in pretty much all cases assert should be used for typechecking only if there's some sort of semantic guarantee not captured by the typesystem that a given type will never appear and if it does it's an error that we want to catch sooner rather than later. Custom exceptions are for when there are non-typing errors and I can't think of a case when casts would be preferred except maybe for not having another statement. I understand it's not idiomatic but in a partially typed codebase like angr it seems way better to use asserts.